src/Sarif.Viewer.VisualStudio/CodeFlowToTreeConverter.cs

```diff
@@ -22,6 +23,18 @@ internal static List<CallTreeNode> Convert(CodeFlow codeFlow)
 
                 foreach (ThreadFlowLocation location in threadFlow.Locations)
                 {
+                    ArtifactLocation artifactLocation = location.Location.PhysicalLocation?.ArtifactLocation;
```

Comment:
>. [](start = 73, length = 1)

Do we need to conditionally access `,PhysicalLocation` as well? Is it possible that `Location` contains only a `LogicalLocation`?

Reply:
For sure. Let's please capture this critical test asset scenario, which is a comprehensive result (i.e., one with a threadflow) that is entirely driven by logical locations, i.e., no physical location data available).

---
In reply to: [416794903](https://github.com/microsoft/sarif-visualstudio-extension/pull/134#discussion_r416794903) [](ancestors = 416794903)

Reply:
Done. Added to [microsoft/tutorials#8](https://github.com/microsoft/sarif-tutorials/issues/8): "[ ] CodeFlow driven entirely by logical locations."

---
In reply to: [416796465](https://github.com/microsoft/sarif-visualstudio-extension/pull/134#discussion_r416796465) [](ancestors = 416796465,416794903)